### ISSUE TITLE
Pass through token update options in `Ruler.prototype._animateMovement` and `_animateSegment`

### ISSUE THREAD

**Issue by caewok** [foundryvtt/foundryvtt#10416]
### User Experience

When moving a token with the ruler, the ruler necessarily updates the token position. But it is currently not possible to pass through options to the `preUpdateToken` and `updateToken` hook. My particular use case is making token easing work more smoothly with ruler segments, so I end up wanting to pass through some information about the ruler segments. Others might want to pass through other options to affect how the token animation works, for example. 

Basically, all that would be needed here is a path-through options object:

```js
async _animateMovement(token, tokenUpdateOptions) {
    ...
    for ( const segment of this.segments ) {
        ...
        await this._animateSegment(token, segment, adjustedDestination, tokenUpdateOptions);
        ...
    }
}

async _animateSegment(token, segment, destination, tokenUpdateOptions) {
    await token.document.update(destination, tokenUpdateOptions);
    ...
}
```

**Comment by aaclayton** [comment]
Good suggestion and something I think we can easily add. Looks like @dev7355608 has already picked this up.
